saleor/checkout/core.py

```diff
@@ -46,85 +46,46 @@ def __iter__(self):
         return iter(self.steps)
 
     def generate_steps(self, cart):
+        shipping_address = None
         self.cart = cart
-        self.billing = BillingAddressStep(
-            self.request, self.get_storage('billing'))
-        self.steps.append(self.billing)
+
         if self.is_shipping_required():
-            self.shipping = ShippingStep(
-                self.request, self.get_storage('shipping'),
-                self.cart, billing_address=self.billing_address)
+            self.shipping = ShippingAddressStep(
+                self.request, self.storage['shipping_address'], checkout=self)
+            shipping_address = self.shipping.address
             self.steps.append(self.shipping)
+            self.delivery = ShippingMethodStep(
```

Comment:
I don't like having `self.shipping` and `self.delivery` as both seem to mean the same thing. Explicitly call them `shipping_address` and `shipping_method`.